hadoop-hdds/tools/src/main/java/org/apache/hadoop/hdds/scm/cli/container/ListSubcommand.java

```diff
@@ -90,11 +95,35 @@ public class ListSubcommand extends ScmSubcommand {
   }
 
 
-  private void outputContainerInfo(ContainerInfo containerInfo)
-      throws IOException {
-    // Print container report info.
+  private void outputContainerInfo(ContainerInfo containerInfo) throws IOException {
+    // Original behavior - just print the container JSON
     System.out.println(WRITER.writeValueAsString(containerInfo));
   }
+  
+  private void outputContainerInfoAsJsonMember(ContainerInfo containerInfo, boolean isFirst,
+      boolean isLast) throws IOException {
+    // JSON array format with proper brackets and commas
+    if (isFirst) {
+      // Start of array
+      System.out.print("[");
```

Comment:
Let's use Jackson's `SequenceWriter` for this similar to what is done in #7944. We may need to duplicate some minor changes to `JsonUtils` here but that should be ok and we can resolve it on the next reverse merge into reconciler branch.

Reply:
Will look into SequenceWriter